ComponentKit/DataSources/CKCollectionViewTransactionalDataSource.mm

```diff
@@ -114,16 +118,32 @@ - (void)transactionalComponentDataSource:(CKTransactionalComponentDataSource *)d
         break;
     }
     
+    // If none of the cells changed size, we can remount the updated cells directly
+    // without notifying the collection view
+    BOOL sizeChanged = NO;
+    for (NSIndexPath *indexPath in changes.updatedIndexPaths) {
+      CKTransactionalComponentDataSourceItem *oldItem = [_currentState objectAtIndexPath:indexPath];
+      CKTransactionalComponentDataSourceItem *newItem = [state objectAtIndexPath:indexPath];
+      sizeChanged = !CGSizeEqualToSize(oldItem.layout.size, newItem.layout.size);
+      if (sizeChanged)
+        break;
+    }
+    
     void (^applyUpdatedState)(CKTransactionalComponentDataSourceState *) = ^(CKTransactionalComponentDataSourceState *updatedState) {
-      [_collectionView performBatchUpdates:^{
+      if (sizeChanged) {
```

Comment:
This check doesn't seem quite right. What if there are no size changes in any of the updates but there are insertions or removals? I imagine we would also want to check `changesIncludeNonUpdates`:

```
if (changesIncludeNonUpdates || (changesIncludeOnlyUpdates && updatesWithBoundsChange)) {
```

You could probably get way with omitting the `changesIncludeOnlyUpdates` check but I think being explicit is helpful.